src/activities/morse_code/MorseCode.qml

```diff
@@ -0,0 +1,540 @@
+                              { label: "1" },
+                              { label: "2" },
+                              { label: "3" },
+                              { label: "4" },
+                              { label: "5" },
+                              { label: "6" },
+                              { label: "7" },
+                              { label: "8" },
+                              { label: "9" }
+                           ],
+                           [
+                              { label: "A" },
+                              { label: "B" },
+                              { label: "C" },
+                              { label: "D" },
+                              { label: "E" },
+                              { label: "F" },
+                              { label: "G" },
+                              { label: "H" },
+                              { label: "I" },
+                              { label: "J" },
+                              { label: "K" },
+                              { label: "L" },
+                              { label: "M" }
+                           ],
+                           [
+                              { label: "N" },
+                              { label: "O" },
+                              { label: "P" },
+                              { label: "Q" },
+                              { label: "R" },
+                              { label: "S" },
+                              { label: "T" },
+                              { label: "U" },
+                              { label: "W" },
+                              { label: "X" },
+                              { label: "Y" },
+                              { label: "Z" },
+                              { label: keyboard.backspace }
+
+                           ]
+                         ]
+
+            }
+
+            function populateMorse() {
+                layout = [ [
+                    { label: "\u00B7" },
+                    { label: "-" },
+                    { label: keyboard.backspace }
+                ] ]
+            }
+
+            onKeypress: {
+                if(!morse_map.visible)
+                    textInput.appendText(text)
+            }
+            onError: console.log("VirtualKeyboard error: " + msg);
+        }
+
+        FirstScreen {
+            id: first_screen
+            visible: true
+        }
+
+        Bar {
+            id: bar
+            anchors.bottom: keyboard.top
+            content: BarEnumContent { value: help | home | level | hint }
+            onHelpClicked: {
+                displayDialog(dialogHelp)
+            }
+            onPreviousLevelClicked: items.previousLevel()
+            onNextLevelClicked: items.nextLevel()
+            onHomeClicked: activity.home()
+            onHintClicked: {
+                morse_map.visible = !morse_map.visible
+                keyboard.visible = !morse_map.visible & ApplicationSettings.isVirtualKeyboard
```

Comment:
No, I intentionally used bitwise AND to make the code for toggling keyboard on and off with morse-map simple.